### PR TITLE
Add vagrant scripts

### PR DESCRIPTION
**Pull request by ccraluca**
In this PR are two commits. 
The first will add some scripts for Jenkins Shared Libray in Vagrant project.
The second will add documentation for each scrippt.

Signed-off-by: Chis Raluca [raluca.chis@analog.com](mailto:raluca.chis@analog.com)

### CODE REVIEW

File: vars/setup_jenkins_agent.groovy [hunk 1]
```diff
@@ -0,0 +1,5 @@
+def call() {
+    sh 'wget https://gist.githubusercontent.com/ccraluca/1a031a1bd71ede37bac01c10c66197c3/raw/fdd341d12263a2c3ebed12ac685f6d1ffe351c1f/jenkins_win_vm_setup.ps1 -O jenkins_win_vm_setup.ps1'
+    sh 'vagrant upload jenkins_win_vm_setup.ps1 "C:\"'
+    sh 'vagrant winrm --command "C:\\jenkins_win_vm_setup.ps1 http://10.48.65.12:8080/swarm/swarm-client.jar win-vm http://10.48.65.12:8080/ C:\\jenkins"'
```
Comment (tfcollins) [inline vars/setup_jenkins_agent.groovy:4]: 1. Can this IP address be passed as an argument?
2. Can you change the name of the script from setup_jenkins_agent -> setup_jenkins_agent_on_vagrant_vm

File: vars/setup_vagrant_box.groovy [hunk 2]
```diff
@@ -0,0 +1,5 @@
+def call(){
+    sh 'wget --continue --tries=0 https://artifactory.analog.com:443/artifactory/sdg-vagrant-dev/test_no_updates.box'
+    sh 'vagrant box add win test_no_updates.box --force'
+    sh 'cp /mnt/storage1/jenkins-workspace/users/rchis/Vagrantfile .'
```
Comment (tfcollins) [inline vars/setup_vagrant_box.groovy:4]: This path to the Vagrantfile should be configurable

File: vars/setup_vagrant_box.groovy [hunk 3]
```diff
@@ -0,0 +1,5 @@
+def call(){
+    sh 'wget --continue --tries=0 https://artifactory.analog.com:443/artifactory/sdg-vagrant-dev/test_no_updates.box'
+    sh 'vagrant box add win test_no_updates.box --force'
```
Comment (tfcollins) [inline vars/setup_vagrant_box.groovy:3]: Should the box name 'win' be an argument?

File: vars/get_vagrant_vm_id.groovy [hunk 4]
```diff
@@ -0,0 +1,11 @@
+def call(){
+    result = sh(returnStdout: true ,script: "vagrant global-status").trim()
+    lines = result.split("\n")
+    len = lines.length-8
+    for(int i = 2;i<len;i++) {
+        if (lines[i].contains("rchis/SharedLibraries_Test")){
```
Comment (tfcollins) [inline vars/get_vagrant_vm_id.groovy:6]: Is this the VM id "rchis/SharedLibraries_Test" ? Should this be an argument?

File: vars/setup_vagrant_box.groovy (hunk 3 above)
Comment (ccraluca) [reply vars/setup_vagrant_box.groovy:3]: Yes, the box name and Vagrantfile path and other specific data will be passed as parameters

File: vars/get_vagrant_vm_id.groovy (hunk 4 above)
Comment (ccraluca) [reply vars/get_vagrant_vm_id.groovy:6]: No, here the function checks for a VM inside rchis/SharedLibraries_Test workspace, so in this way will be sure that we will not use a VM that is running for other Jeknins job

**Comment by tfcollins** [comment]
Is there a main function that wraps most of this logic for end users? For examples something like this:

```groovy
node {
    step('Run VM test') {
        vagrant('win') {
            bat 'hostname'
            ....
        }
    }
}
```

where the `vagrant` function above handles the state for you?

File: vars/setup_jenkins_agent.groovy (hunk 1 above)
Comment (ccraluca) [reply vars/setup_jenkins_agent.groovy]: IP passed as an argument and name of the script changed.

**Comment by ccraluca** [comment]
V3: in this set of changes the documentation was updated according to previous changes and an example Vagrantfile was added for `run_closure()` function

**Comment by ccraluca** [comment]
V4: example for end user added

**Review by tfcollins — CHANGES_REQUESTED** [review]

**Review by tfcollins — APPROVED** [review]

**Pull request merged**